nixos/modules/services/web-apps/shiny-server.nix

```diff
@@ -0,0 +1,129 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.shiny-server;
+
+  generatedConfig = pkgs.writeText "shiny-server.conf" ''
+    run_as ${cfg.user};
+    server {
+      listen ${toString cfg.port};
+      location / {
+        site_dir ${cfg.siteDir};
+        log_dir ${cfg.logDir};
+        bookmark_state_dir ${cfg.stateDir};
+        directory_index ${boolToString cfg.directoryIndex};
```

Comment:
Are there other configuration options that might be applicable here?

Reply:
Some, but I have a hard time imagining when they would be useful. Added another config variable to inject more options.